packages/medusa-payment-paypal/src/services/paypal-provider.js

```diff
@@ -284,11 +282,17 @@ class PayPalProviderService extends PaymentService {
   }
 
   /**
-   * Cancels payment for Stripe payment intent.
-   * @param {object} paymentData - payment method data from cart
+   * Cancels payment for paypal payment.
+   * @param {Payment} payment - payment object
    * @returns {Promise<object>} canceled payment intent
    */
   async cancelPayment(payment) {
+    const order = await this.retrievePayment(payment.data)
+    const isAlreadyCanceled = order.status === "VOIDED" || (order.status === "COMPLETED" && !!order.invoice_id)
```

Comment:
nit(non-blocking): It's not clear why the second case here defines an already canceled payment. Would be great to add a comment above

Reply:
What do you think if I split it in two if(isCanceled || isCanceledAndFullyRefund) that way it is more clear and does not require a comment since the code is self describing itself ?

Reply:
Sounds good!